### PR TITLE
Usage & Limits section

### PR DESCRIPTION
**Pull request by vinibrsl**
### Changes

This pull request add a "Usage & Limits" section to the user settings page. I created a new module called `Plausible.Billing.Quota` to group usage and limits functions, and moved some functions from `Plausible.Billing.Plans` to this new module.

Reviewing commit by commit may help as some files were moved around.

### Preview

Screenshot 2023-08-23 at 09 44 12

### Tests
- [X] Automated tests have been added

### Changelog
- [X] This PR does not make a user-facing change

### Documentation
- [X] This change does not need a documentation update

### Dark mode
- [X] The UI has been tested both in dark and light mode

### CODE REVIEW

File: lib/plausible_web/templates/auth/user_settings.html.heex [hunk 1]
```diff
@@ -115,48 +123,36 @@
...
+        <PlausibleWeb.Components.Billing.usage_and_limits_row
+          title="Total billable pageviews (last 30 days)"
+          usage={@total_pageview_usage}
+          limit={@total_pageview_limit}
+        />
+        <PlausibleWeb.Components.Billing.usage_and_limits_row
+          pad
+          title="Pageviews"
+          usage={@pageview_usage}
+          class="font-normal text-gray-600 dark:text-gray-100"
```
Comment (RobertJoonas) [inline lib/plausible_web/templates/auth/user_settings.html.heex:141]: I think `text-gray-500` would look better. Right now, to me at least, the two text colours look different, but not different enough 😀

I also think that the padded `Pageviews` and `Custom events` should have a different colour on dark mode. Right now they are the same in darkmode:

image

...but different in lightmode

**Review by RobertJoonas — APPROVED** [review]
Nice cleanup on the HTML! 👍

File: lib/workers/check_usage.ex [hunk 2]
```diff
@@ -140,13 +140,13 @@ defmodule Plausible.Workers.CheckUsage do
   end
 
   defp check_site_limit(subscriber) do
```
[the remark this answers is not in the document]
Comment (vinibrsl) [reply lib/workers/check_usage.ex]: Good call! Renamed in 024e7c1534c918ba389136d07ecf0ceb31109a93

File: lib/plausible_web/templates/auth/user_settings.html.heex (hunk 1 above)
Comment (vinibrsl) [reply lib/plausible_web/templates/auth/user_settings.html.heex]: Good call! I changed the color for dark mode and made it lighter for light mode.

**Pull request merged**